apiml-security-common/src/main/java/org/zowe/apiml/security/common/filter/CategorizeCertsFilter.java

```diff
@@ -38,28 +46,89 @@ public class CategorizeCertsFilter extends OncePerRequestFilter {
     private static final String ATTRNAME_CLIENT_AUTH_X509_CERTIFICATE = "client.auth.X509Certificate";
     private static final String ATTRNAME_JAVAX_SERVLET_REQUEST_X509_CERTIFICATE = "javax.servlet.request.X509Certificate";
     private static final String LOG_FORMAT_FILTERING_CERTIFICATES = "Filtering certificates: {} -> {}";
+    private static final String CLIENT_CERT_HEADER = "Client-Cert";
 
+    @InjectApimlLogger
+    private final ApimlLogger apimlLog = ApimlLogger.empty();
+
+    @Getter
     private final Set<String> publicKeyCertificatesBase64;
 
-    public Set<String> getPublicKeyCertificatesBase64() {
-        return publicKeyCertificatesBase64;
-    }
+    private final CertificateValidator certificateValidator;
 
     /**
      * Get certificates from request (if exists), separate them (to use only APIML certificate to request sign and
      * other for authentication) and store again into request.
+     * If authentication via certificate in header is enabled, get client certificate from the Client-Cert header
+     * only if the request certificates are all trusted and validated by {@link CertificateValidator} service.
+     * The client certificate is then stored in a separate custom request attribute. The APIML certificates stays
+     * in the default attribute.
      *
      * @param request Request to filter certificates
      */
     private void categorizeCerts(ServletRequest request) {
         X509Certificate[] certs = (X509Certificate[]) request.getAttribute(ATTRNAME_JAVAX_SERVLET_REQUEST_X509_CERTIFICATE);
         if (certs != null) {
+            if (certificateValidator.isForwardingEnabled() && certificateValidator.isTrusted(certs)) {
+                Optional<Certificate> clientCert = getClientCert((HttpServletRequest) request);
+                if (clientCert.isPresent()) {
+                    // add the client certificate to the certs array
+                    log.debug("Found client certificate in header, adding it to the request.");
```

Comment:
it would help to see which certificate was found, maybe print DN or CN

Reply:
Added subject DN to the debug message.